### PR TITLE
Removing the need to call the createOrdersTableHandler to create the …

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
Issue #2 , Issue #4 

*Description of changes:*
Removing the need to call the createOrdersTableHandler to create the DynamoDB table.
Instead the DynamoDB table is now created in the template.yaml
The table is also created with the on-demand billing type to reduce the risk of incurring costs with the sample.
The functions currently time out with the default MemorySize, increasing this to 512MB.
Adding the correct IAM permissions to the functions to allow them access to the DynamoDB table.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: template.yaml [hunk 1]
```diff
@@ -69,24 +82,26 @@ Resources:
...
-      CodeUri: target/aws-sam-java-rest-1.0.0.jar
-      Handler: com.amazonaws.handler.CreateOrdersTableHandler::handleRequest
-      Runtime: java8
-      Events:
-        CreateOrdersTable:
-          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
-          Properties:
-            Path: /_create_orders_table
-            Method: post
+      TableName: sampleJavaRestTable
```
Comment (SethiShubham) [inline template.yaml]: can you name this to sampleJavaRestOrdersTable instead.

File: template.yaml [hunk 2]
```diff
@@ -21,6 +22,9 @@ Resources:
       CodeUri: target/aws-sam-java-rest-1.0.0.jar
       Handler: com.amazonaws.handler.GetOrderHandler::handleRequest
       Runtime: java8
+      Policies:
+        - DynamoDBCrudPolicy:
```
Comment (kapilsingh5050) [inline template.yaml]: We should use read-only policy (DynamoDBReadPolicy) for read operations

File: template.yaml [hunk 3]
```diff
@@ -69,24 +82,26 @@ Resources:
...
-      CodeUri: target/aws-sam-java-rest-1.0.0.jar
-      Handler: com.amazonaws.handler.CreateOrdersTableHandler::handleRequest
-      Runtime: java8
-      Events:
-        CreateOrdersTable:
-          Type: Api # More info about API Event Source: https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#api
-          Properties:
-            Path: /_create_orders_table
-            Method: post
+      TableName: sampleJavaRestOrdersTable
```
Comment (kapilsingh5050) [inline template.yaml]: What was the name of this table earlier? Changing the name would be sort of a backward incompatible change for existing users of this repository, so I think we should avoid that

File: template.yaml [hunk 4]
```diff
@@ -69,24 +82,26 @@ Resources:
...
+      AttributeDefinitions:
+        - AttributeName: orderId
+          AttributeType: S
+      KeySchema:
+        - AttributeName: orderId
+          KeyType: HASH
+      BillingMode: PAY_PER_REQUEST
```
Comment (kapilsingh5050) [inline template.yaml]: This creates 4 partitions by default on service side which I think would be wasted capacity. We should use a default of Provisioned mode with 5 WCU and 5 RCU. Or keep it same as what it was before

File: template.yaml (hunk 2 above)
Comment (msailes) [reply template.yaml]: agreed

File: template.yaml (hunk 3 above)
Comment (msailes) [reply template.yaml]: I think this is unlikely. The table was named like this.

`return Optional.ofNullable(System.getenv("TABLE_NAME")).orElse("order_table");`

'order_table' is also referenced in each of the environment json files. 
src/test/resources/test_environment_linux.json

I'll change the name back to the original.

File: template.yaml (hunk 4 above)
Comment (msailes) [reply template.yaml]: I disagree.

I think the pay per request model means there is less risk for the customer being charged when they've accidentally left the sample deployed.

**Comment by msailes** [comment]
The unit and integration tests pass, however because the rest endpoint to create a table has been removed the python script doesn't work unless you manual create a table first.

I have removed this rest endpoint because I don't think it's a good idea for a sample. 

Could we consider booting the local dynamodb instance with docker-compose and creating the table as part of that process?

Similiar to this - https://github.com/microservices-demo/orders-aws/blob/master/docker-compose.yml

File: template.yaml (hunk 3 above)
Comment (kapilsingh5050) [reply template.yaml]: I think this change is still remaining

**Comment by kapilsingh5050** [comment]
> The unit and integration tests pass, however because the rest endpoint to create a table has been removed the python script doesn't work unless you manual create a table first.
> 
> I have removed this rest endpoint because I don't think it's a good idea for a sample.
> 
> Could we consider booting the local dynamodb instance with docker-compose and creating the table as part of that process?
> 
> Similiar to this - https://github.com/microservices-demo/orders-aws/blob/master/docker-compose.yml

Yeah let's do this

**Pull request closed without merging**